connect/proxy/config_test.go

```diff
@@ -143,23 +142,17 @@ func TestAgentConfigWatcherSidecarProxy(t *testing.T) {
 			},
 		},
 	}
-
 	require.Equal(t, expectCfg, cfg)
 
 	// Now keep watching and update the config.
-	go func() {
```

Comment:
Correct me if I'm wrong on this! This test isn't _too_ flaky but I have seen it flake recently so wanted to fix it.

The watch is established in a background goroutine and the first assertion proves that the watcher is active so there is no reason for the update to happen in a racy goroutine.

Note that this does not completely remove the race condition as the first call to testGetConfigValTimeout could time out before a config is returned.

Reply:
The original issue I wanted to fix was that `require.NoError` within the goroutine could fire and make go test angry (causing completely unrelated tests to abort). While I was looking for ways to fix it, I figured the registration didn't need to happen in a goroutine.